### DISCUSSION TITLE
New add-on: Script Execution On Mindmap Open

### DISCUSSION THREAD

**Discussion by macmarrum** [freeplane/freeplane#688] [General]
# Script Execution On Mindmap Open

Freeplane has the functionality to execute Groovy scripts when the application starts. It's known as "init scripts" → https://docs.freeplane.org/?search=init

This add-on extends the functionality with the ability to execute scripts when a mindmap is opened, i.e. individually for each mindmap.

To execute a script for a mindmap, in your root node add an attribute named "scriptOnMapOpen", then use `Tools->Edit script...` to enter the Groovy code.

_Note: The attribute name must start with "scriptOnMapOpen" but can have any suffix_

## Download

[**ScriptExecutionOnMindmapOpen.addon.mm**](https://github.com/macmarrum/freeplane-ScriptExecutionOnMindmapOpen/releases/)

## Example

[ScriptExecutionOnMindmapOpen-example.mm.zip](https://github.com/freeplane/freeplane/files/9641790/ScriptExecutionOnMindmapOpen-example.mm.zip)

**Comment by euu2021** [comment]
Nice!👍 

I will test it. It is the foundation to the creation of many interesting scripts.

**Comment by EdoFro** [comment]
It looks nice!!

I think I have some use cases to play with it.

thanks!

**Comment by macmarrum** [comment]
# Script Execution On Mindmap Open - v2022.11.01

A new version of the add-on has been released. 
It introduces an option in the preferences to enable, disable or ask if scriptOnMapOpen is to be executed.

The option is set to "ask" by default, so that when opening mindmaps with unknown content of scriptOnMapOpen, User can opt not to execute the script(s).

## Download
[ScriptExecutionOnMindmapOpen.addon.mm](https://github.com/macmarrum/freeplane-ScriptExecutionOnMindmapOpen/releases/)

## Example (updated)
[ScriptExecutionOnMindmapOpen-example.mm.zip](https://github.com/freeplane/freeplane/files/9912408/ScriptExecutionOnMindmapOpen-example.mm.zip)

**Reply by bepolymathe** [reply]
Out of curiosity, what do you use it for? An example?

**Reply by euu2021** [reply]
I use it to automatically minimize all nodes, as I explained at https://github.com/freeplane/freeplane/discussions/401#discussioncomment-2845218
